packages/contracts-bedrock/test/kontrol/deployment/DeploymentSummary.t.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Libraries
+import { Constants } from "src/libraries/Constants.sol";
+
+// Target contract dependencies
+import { L2OutputOracle } from "src/L1/L2OutputOracle.sol";
+import { SystemConfig } from "src/L1/SystemConfig.sol";
+import { SuperchainConfig } from "src/L1/SuperchainConfig.sol";
+import { OptimismPortal } from "src/L1/OptimismPortal.sol";
+import { DeploymentSummary } from "../proofs/utils/DeploymentSummary.sol";
+import { OptimismPortal_Test } from "test/L1/OptimismPortal.t.sol";
+
+/// @dev Contract testing the deployment summary correctness
+contract DeploymentSummary_Test is DeploymentSummary, OptimismPortal_Test {
+    /// @notice super.setUp is not called on purpose
+    function setUp() public override {
+        // Recreate Deployment Summary state changes
+        DeploymentSummary deploymentSummary = new DeploymentSummary();
+        deploymentSummary.recreateDeployment();
+
+        // Set summary addresses
+        optimismPortal = OptimismPortal(payable(OptimismPortalProxyAddress));
+        superchainConfig = SuperchainConfig(SuperchainConfigProxyAddress);
+        l2OutputOracle = L2OutputOracle(L2OutputOracleProxyAddress);
+        systemConfig = SystemConfig(SystemConfigProxyAddress);
+
+        // Set up utilized addresses
+        depositor = makeAddr("depositor");
+        alice = makeAddr("alice");
+        bob = makeAddr("bob");
+        vm.deal(alice, 10000 ether);
+        vm.deal(bob, 10000 ether);
+    }
+
+    /// @dev Skips the first line of `super.test_constructor_succeeds` because
+    ///      we're not exercising the `Deploy` logic in these tests. However,
+    ///      the remaining assertions of the test are important to check
+    function test_constructor_succeeds() external override {
+        /* OptimismPortal opImpl = OptimismPortal(payable(deploy.mustGetAddress("OptimismPortal"))); */
```

Comment:
Is this dead code?

Reply:
It's commented to highlight the only difference with `super.test_constructor_succeeds`, as stated in the function comment. If it is not clear enough let me know and I can reword it better!